reactor-core/src/main/java/reactor/util/context/ContextView.java

```diff
@@ -0,0 +1,136 @@
+/*
+ * Copyright (c) 2011-Present VMware Inc. or its affiliates, All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *        https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package reactor.util.context;
+
+import java.util.Map;
+import java.util.NoSuchElementException;
+import java.util.Optional;
+import java.util.stream.Stream;
+
+import reactor.util.annotation.Nullable;
+
+/**
+ * A read-only view of a collection of key/value pairs that is propagated between components
+ * such as operators via the context protocol. Contexts are ideal to transport orthogonal
+ * information such as tracing or security tokens.
+ * <p>
+ * {@link Context} is an immutable variant of the same key/value pairs structure which exposes
+ * a write API that returns new instances on each write.
+ *
+ * @author Simon Baslé
+ */
+public interface ContextView {
```

Comment:
nit: could use some auto-formatting of methods' implementations

Reply:
done